redash/query_runner/presto.py

```diff
@@ -93,11 +98,17 @@ def get_schema(self, get_stats=False):
         return list(schema.values())
 
     def run_query(self, query, user):
+        enable_user_impersonation = self.configuration.get("user_impersonation")
+        principle_username = None
+        if enable_user_impersonation and user is not None:
+            principle_username = user.email.split('@')[0]
+
```

Comment:
Maybe I misunderstand the use-case here. Would it not make sense to allow arbitrary user selection in the data source setup screen? Since DS setup is restricted to account admins.

```suggestion
        enable_user_impersonation = (
            user.email.split("@")[0]
            if user is not None and self.configuration.get("user_impersonation", False)
            else None
        )
```

Reply:
So there are a number of parts for this process.

The primary use case for this is the need to have the user executing the query be passed through to Presto, so that Authorization rules and auditing can be done against the person actually running the query.

Internally Presto separates Authentication from Authorization. It also implements the ability for User Impersonation so that a service account with the right access configured in Presto can authenticate then tell Presto to run the query as the principal user. 

The internal behavior within Presto in this situation is that the query will be executed using the defined Resource groups for the principal user, and if they are using Starburst Presto leverage the Ranger Security Access control to determine if the principal user even has access to the resources they are attempting to query.

Because of these two parts, the general expectation is that the user who is logged in, is the person who should be executing the query. That is why it is a boolean setting to turn on User impersonation, and why we are extracting the username from the email address. 

If the desire was to run as a single user, then there would be no need for User Impersonation, as you could just leverage the account configured in the configuration page for authentication.

I hope this helps clarify the use case.

Reply:
Absolutely. This is gold that we can add to the documentation. Thank you for explaining.